packages/resolve/test/request-resolver.spec.ts

```diff
@@ -391,4 +395,96 @@ describe('request resolver', () => {
+      });
+      const resolveRequest = createRequestResolver({ fs });
+
+      expect(resolveRequest('/', './file')).to.be.resolvedTo('/file-browser.js');
+    });
+
+    it('supports remapping of package to package', () => {
+      const fs = createMemoryFs({
+        'package.json': stringifyPackageJson({ browser: { fs: 'browser-fs' } }),
+        node_modules: {
+          'browser-fs': {
+            'package.json': stringifyPackageJson({ main: './file.js' }),
+            'file.js': EMPTY,
+          },
+        },
+      });
+      const resolveRequest = createRequestResolver({ fs });
+
+      expect(resolveRequest('/', 'fs')).to.be.resolvedTo('/node_modules/browser-fs/file.js');
+    });
+
+    it('supports remapping of package to false (empty object)', () => {
+      const fs = createMemoryFs({
+        'package.json': stringifyPackageJson({ browser: { fs: false } }),
+      });
+      const resolveRequest = createRequestResolver({ fs });
+
+      expect(resolveRequest('/', 'fs')).to.be.resolvedTo(false);
+    });
+
+    it('supports remapping of a local file to false (empty object)', () => {
+      const fs = createMemoryFs({
+        'package.json': stringifyPackageJson({ browser: { './some-file.js': false } }),
+        'some-file.js': EMPTY,
+      });
+      const resolveRequest = createRequestResolver({ fs });
+
+      expect(resolveRequest('/', './some-file')).to.be.resolvedTo(false);
+    });
+
+    it('supports remapping of package to a relative file', () => {
+      const fs = createMemoryFs({
+        'package.json': stringifyPackageJson({ browser: { fs: './browser-fs' } }),
+        'browser-fs.js': EMPTY,
+      });
+      const resolveRequest = createRequestResolver({ fs });
+
+      expect(resolveRequest('/', 'fs')).to.be.resolvedTo('/browser-fs.js');
+    });
+
+    it('supports packages remapping their entrypoint', () => {
+      const fs = createMemoryFs({
+        node_modules: {
+          'some-package': {
+            'package.json': stringifyPackageJson({ main: './file.js', browser: { './file': './file-browser' } }),
+            'file.js': EMPTY,
+            'file-browser.js': EMPTY,
+          },
+        },
+      });
+      const resolveRequest = createRequestResolver({ fs });
+
+      expect(resolveRequest('/', 'some-package')).to.be.resolvedTo('/node_modules/some-package/file-browser.js');
+      expect(resolveRequest('/', 'some-package/file')).to.be.resolvedTo('/node_modules/some-package/file-browser.js');
+      expect(resolveRequest('/node_modules/some-package', './file')).to.be.resolvedTo(
+        '/node_modules/some-package/file-browser.js'
+      );
+    });
+
+    it('ignores re-mapping when source/target are missing/invalid', () => {
+      const fs = createMemoryFs({
+        'package.json': stringifyPackageJson({
+          browser: {
+            './file': (123 as unknown) as string,
+            './missing-source': './missing-target',
+            './another-missing': './file',
```

Comment:
why are these needed in this example?

Reply:
there are many combinations where stuff can fail (source is missing, target is missing, both are missing). instead of creating 3 different tests, I'm testing all similar combinations within this one.

Reply:
I added explicit assertions now that ensure the above don't cause accidental re-mapping.
Previously, it **implicitly** verified that the other two re-mappings get ignored, while the third one works.

Nice catch.